dulwich/porcelain.py

```diff
@@ -2098,6 +2099,123 @@ def checkout_branch(repo, target: Union[bytes, str], force: bool = False) -> Non
                 dir_path = os.path.dirname(dir_path)
 
 
+# --- Sparse Checkout Commands ---
+
+
+def sparse_checkout(repo, patterns=None, force=False):
+    """Perform a sparse checkout by updating the index and working tree
+    based on inclusion/exclusion patterns, using skip-worktree bits in
+    the index's extended_flags field (as supported by patched Dulwich).
+
+    Args:
+      repo: A path to a repository or a Repo instance
+      patterns: A list of gitignore-style patterns
+      force: If True, destructive operations proceed even if local changes exist
+
+    Raises:
+      CheckoutError: If local modifications would be lost (force=False)
+      Error: If no patterns can be determined or an I/O error occurs
+    """
+    repo = Repo(repo) if not isinstance(repo, Repo) else repo
+
+    info_dir = os.path.join(repo.path, ".git", "info")
+    if not os.path.exists(info_dir):
+        os.makedirs(info_dir)
+    sparse_file = os.path.join(info_dir, "sparse-checkout")
```

Comment:
since these are internal repository details, they should really be handled by the Repo class rather than porcelain.

Reply:
Yeah I think `get`/`set` + `_sparse_checkout_patterns` would be a nice way to abstract the following if/else block too, so this would become internal to the Repo